src/PaymentExporter.php

```diff
@@ -17,10 +17,13 @@ class PaymentExporter {
    *   An associative array of data for this payment.
    */
   public function toJson(\Payment $payment) {
+    $currency = currency_load($payment->currency_code);
+    $subunits = $currency->subunits ?: 1;
     $data['pid'] = $payment->pid;
     $status = $payment->getStatus();
     $data['currency_code'] = $payment->currency_code;
     $data['total_amount'] = $payment->totalAmount(TRUE);
+    $data['total_amount_subunits'] = (int) round($payment->totalAmount(TRUE) * $subunits);
```

Comment:
Not sure why it needs a "total_amount" *and* "total_amount_subunits" but if that’s what logcrm expects, this module should provide it :shrug:

Reply:
I tried to figure out if just using `round()` is OK, because there are different rounding rules around currencies; but they seem to only apply to money as cash. Best information I found was https://unicode-org.github.io/cldr-staging/charts/37/supplemental/detailed_territory_currency_information.html#format_info

Reply:
Example: a total amount of CHF 1,025 would be 103 subunits but 1 Franken and 5 Rappen in cash (1,05).

This probably answers the question about the subunits as well... :grin:

Reply:
I tried not to make a rocket sience of it. An integer of the smallest currency unit seemed to be the best way to represent amounts in a computer though.

Reply:
But to answer the original question:

- `total_amount` was there before this change. Removing it would break backwards compatibility.
- `total_amount_subunits` is what I want to have for logCRM (see [logcrm!11](https://gitlab.more-onion.com/misc/logcrm/-/merge_requests/11))

Reply:
I also do not want to make rocket science out of this, but, still, I think it is good to know about the rounding rules. Or at least that there are different ones and that there are no issue here.

Reply:
In theory, logcrm could still be able to deduct the total amount from the subunits…? :thinking: